core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3226,6 +3226,21 @@ public static ByteString bitNot(ByteString b) {
 
     return new ByteString(result);
   }
+  public static int leftShift(int a, int b) {
+    return a << b;
+  }
+
+  public static long leftShift(long a, int b) {
+    return a << b;
+  }
+
+  public static long leftShift(int a, long b) {
+    return a << b;
+  }
+
+  public static long leftShift(Long a, Long b) {
```

Comment:
will this cover all supported types, like TINYINT?
What is the semantics of shift when the shift amount is negative or larger than the value size?